build.gradle.kts

```diff
@@ -2,29 +2,53 @@ plugins {
     id("java")
     id("org.springframework.boot") version "3.4.3"
     id("io.spring.dependency-management") version "1.1.4"
-    kotlin("jvm") version "1.9.21"
-    kotlin("plugin.spring") version "1.9.21"
 }
 
-
 group = "org.example"
 version = "1.0-SNAPSHOT"
 
 repositories {
     mavenCentral()
 }
 
+sourceSets {
+    val functionalTest by creating {
+        compileClasspath += sourceSets["main"].output
+        runtimeClasspath += sourceSets["main"].output
+    }
+}
+
+val functionalTestImplementation by configurations.getting {
+    extendsFrom(configurations["implementation"])
+}
+
 dependencies {
     implementation("org.springframework.boot:spring-boot-starter")
     implementation("org.springframework.boot:spring-boot-starter-web")
     implementation("org.hibernate:hibernate-validator:8.0.2.Final")
     implementation("javax.inject:javax.inject:1")
 
-    testImplementation("org.springframework.boot:spring-boot-starter-test")
     testImplementation(platform("org.junit:junit-bom:5.10.0"))
     testImplementation("org.junit.jupiter:junit-jupiter")
+
+    functionalTestImplementation("org.springframework.boot:spring-boot-starter-test")
+}
+
+tasks.register<Test>("functionalTest") {
+    description = "Runs functional tests."
+    group = "verification"
+    testClassesDirs = sourceSets["functionalTest"].output.classesDirs
+    classpath = sourceSets["functionalTest"].runtimeClasspath
+    shouldRunAfter("test")
+    useJUnitPlatform()
+    systemProperty("user.language", "en")
```

Comment:
we can use custom .properties file to use our own messages instead of using language dependent messages

see example for `weight` validation